wordpress-mobile/WordPress-Android
# Displays transfer domains webview from domains search screen

Fixes #19383

**Based on:** https://github.com/wordpress-mobile/WordPress-Android/pull/19466

## Description
This PR displays the transfer domains webview when the user taps on the transfer domain button already implemented in the search domain screen with https://github.com/wordpress-mobile/WordPress-Android/pull/19466

## To test
1. Make sure you have enabled the domain_management feature flag.
2. Go to Me -> Domains
3. Tap on "+" button on the toolbar
4. Tap on the "Transfer domain" button at the bottom of the screen
5. **Verify** that the "Transfer Your Domains" page opens
6. **Verify** that the `domain_management_transfer_domain_tapped` tracking event is emitted 

https://github.com/wordpress-mobile/WordPress-Android/assets/304044/d2a12acd-d511-468c-bca6-2258b13d13e4

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added new test cases in `NewDomainSearchViewModelTest`

PR submission checklist:


UI Changes testing checklist:
